lib/shared/drive-constraints.js

```diff
@@ -66,6 +68,66 @@ exports.isSystemDrive = (drive) => {
 };
 
 /**
+ * @summary Check if a drive is source drive
+ * @function
+ * @public
+ *
+ * @description
+ * In the context of Etcher, a source drive is a drive
+ * containing the image.
+ *
+ * @param {Object} drive - drive
+ * @param {Object} image - image
+ * @returns {Boolean} whether the drive is a source drive
+ *
+ *
+ * @example
+ * if (constraints.isSourceDrive({
+ *   device: '/dev/disk2',
+ *   name: 'My Drive',
+ *   size: 123456789,
+ *   protected: true,
+ *   system: true,
+ *   mountpoints: [
+ *     {
+ *       path: '/Volumes/Untitled'
+ *     }
+ *   ]
+ * }, {
+ *   path: '/Volumes/Untitled/image.img',
+ *   size: 1000000000
+ * })) {
+ *   console.log('This drive is a source drive!');
+ * }
+ */
+exports.isSourceDrive = (drive, image) => {
+  const mountpoints = _.get(drive, 'mountpoints', []);
+
+  if (!image || !image.path || _.isEmpty(mountpoints)) {
+    return false;
+  }
+
+  // NodeJS only allows manipulation Windows paths
+  // from Windows and UNIX paths from UNIX, so we
+  // have to manually set/unset `path.sep` to make
+  // this piece of code truly cross-platform.
```

Comment:
Huh? Presumably this is only for the unit-tests though, since Etcher running on Windows will never be passed a Unix-style path, and vice-versa? If that's the case, wouldn't it be better for the unit-tests to be modified to check current-platform paths, rather than 'polluting' this module with mucking around with `path.sep`?